met/src/tools/core/ensemble_stat/ensemble_stat.cc

```diff
@@ -2508,6 +2365,302 @@ void build_outfile_name(unixtime ut, const char *suffix, ConcatString &str) {
+         }
+      }
+
+      // Write RELP counts
+      if(vx_opt.output_flag[i_relp] != STATOutputType_None) {
+
+         pd.compute_relp();
+
+         if(pd.relp_na.sum() > 0) {
+            write_relp_row(shc, &pd,
+               vx_opt.output_flag[i_relp],
+               stat_at, i_stat_row,
+               txt_at[i_relp], i_txt_row[i_relp]);
+         }
+      }
+
+      // Write SSVAR scores
+      if(vx_opt.output_flag[i_ssvar] != STATOutputType_None) {
+
+         pd.ssvar_bin_size = vx_opt.vx_pd.pd[0][0][0].ssvar_bin_size;
+         pd.compute_ssvar();
+
+         // Make sure there are bins to process
+         if(pd.ssvar_bins) {
+
+            // Add rows to the output AsciiTables for SSVAR
+            stat_at.add_rows(pd.ssvar_bins[0].n_bin *
+                             vx_opt.ci_alpha.n());
+
+            if(vx_opt.output_flag[i_ssvar] == STATOutputType_Both) {
+               txt_at[i_ssvar].add_rows(pd.ssvar_bins[0].n_bin *
+                                        vx_opt.ci_alpha.n());
+            }
+
+            // Write the SSVAR data for each alpha value
+            for(j=0; j<vx_opt.ci_alpha.n(); j++) {
+               write_ssvar_row(shc, &pd, vx_opt.ci_alpha[j],
+                  vx_opt.output_flag[i_ssvar],
+                  stat_at, i_stat_row,
+                  txt_at[i_ssvar], i_txt_row[i_ssvar]);
+            }
+         }
+      }
+
+   } // end for i
+
+   // Write PCT counts and scores
+   if(vx_opt.output_flag[i_pct]  != STATOutputType_None ||
+      vx_opt.output_flag[i_pstd] != STATOutputType_None ||
+      vx_opt.output_flag[i_pjc]  != STATOutputType_None ||
+      vx_opt.output_flag[i_prc]  != STATOutputType_None ||
+      vx_opt.output_flag[i_eclv] != STATOutputType_None) {
+      do_pct(vx_opt, pd_all);
+   }
+
+   // Write out the unfiltered ORANK lines for point verification
+   if(is_point_vx &&
+      vx_opt.output_flag[i_orank] != STATOutputType_None) {
+
+      // Set the header column
+      shc.set_obs_thresh(na_str);
+
+      write_orank_row(shc, &pd_all,
+         vx_opt.output_flag[i_orank],
+         stat_at, i_stat_row,
+         txt_at[i_orank], i_txt_row[i_orank]);
+
+      // Reset the observation valid time
+      shc.set_obs_valid_beg(vx_opt.vx_pd.beg_ut);
+      shc.set_obs_valid_end(vx_opt.vx_pd.end_ut);
+   }
+
+   return;
+}
+
+////////////////////////////////////////////////////////////////////////
+
+void do_pct(const EnsembleStatVxOpt &vx_opt, const PairDataEnsemble &pd_ens) {
+   int i, j, k;
+   int i_obs, i_ens, n_vld, n_thr, n_bin;
+   PCTInfo *pct_info = (PCTInfo *) 0;
+   PairDataPoint pd_pnt, pd;
+   ConcatString fcst_var_cs, cs;
+
+   mlog << Debug(2)
+        << "Computing Probabilistic Statistics.\n";
+
+   // Derive a PairDataPoint object from the PairDataEnsemble input
+   pd_pnt.extend(pd_ens.n_obs);
+
+   // Determine the number of climo CDF bins
+   n_bin = (pd_pnt.cmn_na.n_valid() > 0 && pd_pnt.csd_na.n_valid() > 0 ?
```

Comment:
@j-opatz this is the spot in the code whose logic I'd like to discuss. I *think* if the user hasn't defined prob_cat_thresh but has defined climo_mean and climo_stdev along with climo bins, we want to use those thresholds to define/evaluate probabilities. But this is the logic that always gets so confusing!